entrypoint.sh

```diff
@@ -139,21 +140,24 @@ for repository in "${REPOSITORIES[@]}"; do
         echo " "
     done
 
-    cd ${GIT_PATH}
-
-    # push changes
-    echo "Push changes to [${REPO_URL}]"
-    git push $REPO_URL
-    if [ ! -z "$PULL_REQUEST_BRANCH_NAME" -a "$BRANCH_NAME" != "$PULL_REQUEST_BRANCH_NAME" ]; then
-        echo "Creating pull request from [$BRANCH_NAME] into [$PULL_REQUEST_BRANCH_NAME]"
-        jq -n --arg title "File sync from ${GITHUB_REPOSITORY}" --arg head "$BRANCH_NAME" --arg base $PULL_REQUEST_BRANCH_NAME '{title:$title,head:$head,base:$base}' | curl -d @- \
-            -X POST \
-            -H "Accept: application/vnd.github.v3+json" \
-            -u ${USERNAME}:${GITHUB_TOKEN} \
-            --silent \
-            ${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/pulls
+    if [ "$need_push" = true ] ; then
+        cd ${GIT_PATH}
+
+        # push changes
+        echo "Push changes to [${REPO_URL}]"
+        git push $REPO_URL
+        if [ ! -z "$PULL_REQUEST_BRANCH_NAME" -a "$BRANCH_NAME" != "$PULL_REQUEST_BRANCH_NAME" ]; then
+            echo "Creating pull request from [$BRANCH_NAME] into [$PULL_REQUEST_BRANCH_NAME]"
+            jq -n --arg title "File sync from ${GITHUB_REPOSITORY}" --arg head "$BRANCH_NAME" --arg base $PULL_REQUEST_BRANCH_NAME '{title:$title,head:$head,base:$base}' | curl -d @- \
+                -X POST \
+                -H "Accept: application/vnd.github.v3+json" \
+                -u ${USERNAME}:${GITHUB_TOKEN} \
+                --silent \
+                ${GITHUB_API_URL}/repos/${REPO_NAME}/pulls
+        fi
+        cd $TEMP_PATH
```

Comment:
i think this needs to be outside the if statement, it's used to cleanup folders

Reply:
fixed